Firestore/core/src/firebase/firestore/remote/grpc_connection.h

```diff
@@ -80,25 +82,29 @@ class GrpcConnection {
   void Unregister(GrpcCall* call);
 
   /**
-   * For tests only: use a custom root certificate file and the given SSL
-   * target name for all connections. Call before creating any streams or calls.
+   * Don't use SSL, send all traffic unencrypted. Call before creating any
+   * streams or calls.
    */
-  static void UseTestCertificate(absl::string_view certificate_path,
-                                 absl::string_view target_name);
+  static void UseInsecureChannel(const std::string& host);
 
   /**
-   * For tests only: don't use SSL, send all traffic unencrypted. Call before
-   * creating any streams or calls. Overrides a test certificate.
+   * For tests only: use a custom root certificate file and the given SSL
+   * target name for all connections. Call before creating any streams or calls.
    */
-  static void UseInsecureChannel();
+  static void UseTestCertificate(const std::string& host,
+                                 const util::Path& certificate_path,
+                                 const std::string& target_name);
 
  private:
-  struct TestCredentials {
-    std::string certificate_path;
+  static bool HasSpecialConfig(const std::string& host);
+
+  struct HostConfig {
+    util::Path certificate_path;
     std::string target_name;
     bool use_insecure_channel = false;
   };
-  static TestCredentials* test_credentials_;
+  using ConfigByHost = std::unordered_map<std::string, HostConfig>;
```

Comment:
Could these details be entirely private (and with static linkage) in the implementation file?